index.bs

```diff
@@ -286,65 +290,76 @@ When the `[[track]]` of a {{MediaStreamTrackProcessor}} |processor|
-    with |init|.{{MediaStreamTrackGeneratorInit/kind}}.
-4. Return |g|.
+1. Let |source| be a new {{VideoTrackSource}} object.
+1. Let |track| be a new {{MediaStreamTrack}} object, whose {{MediaStreamTrack/kind}} is "video", and whose {{MediaStreamTrack/id}} is a new unique id generated by the user agent.
+1. Initialize |source|.{{VideoTrackSource/track}} to |track|.
+4. Return |source|.
 
-### Attributes ### {#generator-attributes}
+### Attributes ### {#video-source-attributes}
 <dl>
-<dt><dfn attribute for=MediaStreamTrackGenerator>writable</dfn></dt>
-<dd>Allows writing media frames to the {{MediaStreamTrackGenerator}}, which is
-itself a {{MediaStreamTrack}}. When this attribute is accessed for the first time,
-it MUST be initialized with the following steps:
-1. Initialize [=this=].{{MediaStreamTrackGenerator/writable}} to be a new {{WritableStream}}.
-2. <a dfn for="WritableStream">Set up</a> [=this=].{{MediaStreamTrackGenerator/writable}}, with its [=WritableStream/set up/writeAlgorithm=] set to [=writeFrame=] with |this| as parameter, with [=WritableStream/set up/closeAlgorithm=] set to [=closeWritable=] with |this| as parameter and [=WritableStream/set up/abortAlgorithm=] set to [=closeWritable=] with |this| as parameter.
-
-The <dfn>writeFrame</dfn> algorithm is given a |generator| and a |frame| as input. It is defined by running the following steps:
-1. If |generator|.{{MediaStreamTrack/kind}} equals `video` and |frame| is not a {{VideoFrame}} object, return [=a promise rejected with=] a {{TypeError}}.
-2. If |generator|.{{MediaStreamTrack/kind}} equals `audio` and |frame| is not an {{AudioData}} object, return [=a promise rejected with=] a {{TypeError}}.
-3. Send the media data backing |frame| to all live tracks connected to |generator|, possibly including |generator| itself.
-4. Invoke the `close` method of |frame|.
-5. Return  [=a promise resolved with=] undefined.
+<dt><dfn attribute for=VideoTrackSource>writable</dfn></dt>
+<dd>Allows writing video frames to the {{VideoTrackSource}}. When this attribute
+is accessed for the first time, it MUST be initialized with the following steps:
+1. Initialize [=this=].{{VideoTrackSource/writable}} to be a new {{WritableStream}}.
+2. <a dfn for="WritableStream">Set up</a> [=this=].{{VideoTrackSource/writable}}, with its [=WritableStream/set up/writeAlgorithm=] set to [=writeFrame=] with |this| as parameter, with [=WritableStream/set up/closeAlgorithm=] set to [=closeWritable=] with |this| as parameter and [=WritableStream/set up/abortAlgorithm=] set to [=closeWritable=] with |this| as parameter.
+
+The <dfn>writeFrame</dfn> algorithm is given a |source| and a |frame| as input. It is defined by running the following steps:
+1. If |frame| is not a {{VideoFrame}} object, return [=a promise rejected with= a {{TypeError}}.
+1. If |source|.`[[isMuted]]` is false, send the media data backing |frame| to all live tracks sourced from |source|.
+1. Invoke the `close` method of |frame|.
```

Comment:
Seems good enough for now, I believe there might be cases where the close method will be called too late though so that JS could call close() before running that step for instance. Also, we should handle the case of closed frames.